src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -550,15 +549,12 @@ doGetTxLog d txid = do
                       _deltaTableName delta == tableNameBS
 
     tableName = domainTableName d
-    (Utf8 tableNameBS) = tableName
-
-    takeHead [] = []
-    takeHead (a:_) = [a]
+    Utf8 tableNameBS = tableName
 
     readFromPending = do
         ptx <- maybeToList <$> use bsPendingTx
         pb <- use bsPendingBlock
-        let deltas = (ptx ++ [pb]) >>= HashMap.elems . _pendingWrites >>= takeHead . DL.toList
+        let deltas = (ptx ++ [pb]) >>= map NE.toList . HashMap.elems . HashMap.unions . HashMap.elems . _pendingWrites >>= take 1
```

Comment:
At this point I feel you should just write a `do` statement here rather than making this line even longer, TBH.

Reply:
I don't really even know if this line is correct. I'm going to try to tease it out.